scripts/backupmanager.sh

```diff
@@ -0,0 +1,336 @@
+
+# Default values if the environment variables exist
+
+BACKUP_DIR=${BACKUP_PATH} # Dir where the backup files are stored
+# RESTORE_FILE_PATH=${TRIGGER_RESTORE_PATH} # Dir where the restore trigger file with the backup file name is stored
+# BACKUP_FILE=${TRIGGER_RESTORE_FILE} # Path of the backup file trigger
+SAVE_DIR=${GAME_SAVE_PATH} # Dir where the game save files are stored
+BACKUPS_TO_KEEP=${BACKUP_RETENTION_AMOUNT} # Number of backup files to keep
+
+
+# Function to print usage information
+function print_usage() {
+    script_name=$(basename "$0")
+    echo "Usage:"
+    echo "  ${script_name} [backup_directory] -c|--create"
+    # echo "  ${script_name} [backup_directory] -r|--restore [backup_name]"
+    echo "  ${script_name} [backup_directory] -l|--list [num_entries]"
+    echo "  ${script_name} [backup_directory] -cl|--clean [num_to_keep]"
+    echo "  ${script_name} -h|--help"
+    echo ""
+    echo "Options:"
+    echo "  -c, --create                    Create a backup"
+    # echo "  -r, --restore [backup_name]     Restore from a backup"
+    echo "  -l, --list [num_entries]        List the backup files. If num_entries isn't"
+    echo "                                  provided, all backup files will be listed"
+    echo "  -cl, --clean [num_to_keep]      Deletes old backups keeping the num_to_keep"
+    echo "                                  most recent backups. If num_to_keep isn't"
+    echo "                                  provided, keep 30 most recent backups"
+    echo "  -h, --help                      Display this help message"
+    echo ""
+    echo "Arguments:"
+    echo "  backup_directory    (optional) The directory where the backup files are stored."
+    echo "                                 If not provided, the value of the BACKUP_PATH"
+    echo "                                 environment variable will be used if it exists"
+    echo "                                 and is not empty."
+    # echo "  backup_name         (optional) The name of the backup file to restore from."
+    # echo "                                 If not provided, the value of the BACKUP_FILE"
+    # echo "                                 environment variable will be used if it exists"
+    # echo "                                 and is not empty."
+    echo "  num_entries         (optional) The number of backup files to list."
+    echo "                                 If not provided, all backup files will be listed"
+    echo "  num_to_keep         (optional) The number of the most recent backup files to keep."
+    echo "                                 If not provided, the value of the BACKUP_RETENTION_AMOUNT"
+    echo "                                 environment variable will be used if it exists and is not empty."
+    echo "                                 Defaults to the 5 most recent backups otherwise."
+}
+
+
+# Function to parse input arguments
+function parse_arguments() {
+    if [ ${#} -lt 1 ]; then
```

Comment:
This check should have been done before entering the function.
This should be in initializeBackupManager and if nothing is valid show usage.
This function jumping is un-neccesary and disturbs the flow of reading code, because you have to jump every 2 seconds to antoher function, you miss overview and context that way. KISS principle please (Keep it simple and stupid).